File: webview_useragent/android/src/main/kotlin/com/strysk/webview_useragent/webview_useragent/WebviewUserAgentPlugin.kt
package com.strysk.webview_useragent.webview_useragent

import androidx.annotation.NonNull
import android.webkit.WebView

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** WebviewUserAgentPlugin */
class WebviewUserAgentPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  var webView: WebView? = null

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "webview_useragent")
    channel.setMethodCallHandler(this)
    webView = WebView(flutterPluginBinding.applicationContext)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method == "getPlatformUserAgent") {
      val userAgent = webView?.settings?.userAgentString
      result.success(userAgent)
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}

File: webview_useragent_ios/ios/Classes/WebviewUserAgentPlugin.swift
import Flutter
import UIKit
import WebKit

public class WebviewUserAgentPlugin: NSObject, FlutterPlugin {
    var webView: WKWebView = WKWebView()

    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "webview_useragent", binaryMessenger: registrar.messenger())
        let instance = WebviewUserAgentPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformUserAgent":
            webView.evaluateJavaScript("navigator.userAgent;", completionHandler: { (jsResult, error) in
                guard let userAgent = jsResult as? String else { return }
                result(userAgent)
            })

        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
